Rest/ContentUserGateway.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace EzSystems\PlatformUIBundle\Rest;
+
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\Core\Repository\Values\User\UserCreateStruct;
+use eZ\Publish\Core\REST\Common\Input\Dispatcher;
+use eZ\Publish\Core\REST\Common\Message;
+use eZ\Publish\Core\REST\Server\Values\CreatedContent;
+use eZ\Publish\Core\REST\Server\Values\CreatedUser;
+use eZ\Publish\Core\REST\Server\Values\RestContent;
+use eZ\Publish\Core\REST\Server\Values\RestContentCreateStruct;
+use eZ\Publish\Core\REST\Server\Values\RestUser;
+use eZ\Publish\Core\REST\Server\View\AcceptHeaderVisitorDispatcher;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use eZ\Publish\Core\FieldType\User\Value as UserFieldValue;
+
+class ContentUserGateway implements EventSubscriberInterface
+{
+    /** @var \eZ\Publish\Core\REST\Common\Input\Dispatcher */
+    private $restInputDispatcher;
+
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
+
+    /** @var \eZ\Publish\Core\REST\Server\View\AcceptHeaderVisitorDispatcher */
+    private $viewDispatcher;
+
+    public function __construct(Dispatcher $restInputDispatcher, AcceptHeaderVisitorDispatcher $viewDispatcher, Repository $repository)
+    {
+        $this->restInputDispatcher = $restInputDispatcher;
+        $this->repository = $repository;
+        $this->viewDispatcher = $viewDispatcher;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [KernelEvents::REQUEST => ['onKernelRequest', -1000]];
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        if (!$request->attributes->get('is_rest_request')) {
+            return;
+        }
+
+        if (!$contentTypeHeaderValue = $request->headers->get('content-type')) {
```

Comment:
we tend to prefer parentheses for these right?
`if (!($contentTypeHeaderValue = $request->headers->get('content-type'))) {`